smithy-aws-protocol-tests/model/awsJson1_0/defaults.smithy

```diff
@@ -0,0 +1,320 @@
+                    "defaultDocumentBoolean": false,
+                    "defaultDocumentList": ["b"],
+                    "defaultNullDocument": "notNull",
+                    "defaultTimestamp": 2,
+                    "defaultBlob": "aGk=",
+                    "defaultByte": 2,
+                    "defaultShort": 2,
+                    "defaultInteger": 20,
+                    "defaultLong": 200,
+                    "defaultFloat": 2.0,
+                    "defaultDouble": 2.0,
+                    "defaultMap": {"name": "Jack"},
+                    "defaultEnum": "BAR",
+                    "defaultIntEnum": 2
+                }"""
+            params: {
+                defaultString: "bye"
+                defaultBoolean: false
+                defaultList: ["a"]
+                defaultDocumentMap: {name: "Jack"}
+                defaultDocumentString: "bye"
+                defaultDocumentBoolean: false
+                defaultDocumentList: ["b"]
+                defaultNullDocument: "notNull"
+                defaultTimestamp: 2
+                defaultBlob: "hi"
+                defaultByte: 2
+                defaultShort: 2
+                defaultInteger: 20
+                defaultLong: 200
+                defaultFloat: 2.0
+                defaultDouble: 2.0
+                defaultMap: {name: "Jack"}
+                defaultEnum: "BAR"
+                defaultIntEnum: 2
+            }
+        }
+        {
+            id: "AwsJson10ServerPopulatesDefaultsInResponseWhenMissingInParams"
+            documentation: "Server populates default values in response when missing in params."
+            appliesTo: "server"
+            tags: ["defaults"]
+            protocol: awsJson1_0
+            code: 200
+            bodyMediaType: "application/json"
+            headers: {"Content-Type": "application/x-amz-json-1.0"}
+            body: """
+                {
+                    "defaultString": "hi",
+                    "defaultBoolean": true,
+                    "defaultList": [],
+                    "defaultDocumentMap": {},
+                    "defaultDocumentString": "hi",
+                    "defaultDocumentBoolean": true,
+                    "defaultDocumentList": [],
+                    "defaultTimestamp": 0,
+                    "defaultBlob": "YWJj",
+                    "defaultByte": 1,
+                    "defaultShort": 1,
+                    "defaultInteger": 10,
+                    "defaultLong": 100,
+                    "defaultFloat": 1.0,
+                    "defaultDouble": 1.0,
+                    "defaultMap": {},
+                    "defaultEnum": "FOO",
+                    "defaultIntEnum": 1
+                }"""
+            params: {}
+        }
+    ]
+)
+
+operation OperationWithDefaults {
+    input := {
+        @required
+        defaults: Defaults
+
+        topLevelDefault: String = "hi" // Client should ignore default values in input shape
```

Comment:
That should be covered by the `AwsJson10ClientPopulatesDefaultValuesInInput` test

Reply:
It can be useful for implementers to validate their protocol work piecemeal. A separate test would let them verify this specific behavior while not having built all the support for the much larger test case.